rfcs/0019-maintainers.md

```diff
@@ -0,0 +1,197 @@
+---
+feature: maintainers-file
+start-date: 2017-10-28
+author: "Maarten Hoogendoorn (@moretea)"
+co-authors: "@zimbatm"
+related-issues: "-"
+---
+
+# Summary
+[summary]: #summary
+
+In the long term we want to move to a more controlled way of can merge what, and have explicit
+reviewers in place for all files.
+This RFC is part of a larger body of work that is necessary to achieve this goal.
+To enable this, we must be able to map all nixpkgs files to maintainers.
+
+
+# Motivation
+[motivation]: #motivation
+<!--  Why are we doing this? -->
+Currently we have two mechanism in place that explicitly describes who the owner is.
+It either is defined in the `meta.maintainers` in a `stdenv.mkDerivation`, or when it matches a
+pattern in `CODEOWNERS` file.
+
+The `meta.maintainers` only covers the packages, and some NixOS modules and NixOS tests.
+It is not used to determine reviewers on GitHub.
+
+The [`CODEOWNERS`](https://help.github.com/articles/about-codeowners/)
+is only used to help GitHub select a reviewer and is centrally managed in a single file.
+One of the problems is that this loses the data locality, unlike the `.gitignore` files, on which
```

Comment:
How so? Because we have two places we describe maintainers? What about nested `.gitignore` files?